src/compiler/emitter.ts

```diff
@@ -307,6 +312,7 @@ namespace ts {
 
         function setSourceFile(sourceFile: SourceFile) {
             currentSourceFile = sourceFile;
+            statementOffset = getIndexOfFirstNonPrologueDirectives(sourceFile.statements);

```

Comment:
I'd rather we not cache this as it introduces a new entry in the environment record for the closures created for `setSourceFile` and `emitSourceFileWorker`. See [my comment below](https://github.com/Microsoft/TypeScript/pull/14274/files#r103816909) as to how to avoid caching this.